ledger/src/blockstore_db.rs

```diff
@@ -419,7 +419,12 @@ impl Rocks {
         }
         let oldest_slot = OldestSlot::default();
         let column_options = options.column_options.clone();
-        let cf_descriptors = Self::cf_descriptors(&options, &oldest_slot);
+        let detected_cfs = DB::list_cf(&Options::default(), path)
+            .map_err(|err| {
+                warn!("Unable to detect Rocks colums: {err:?}");
+            })
+            .unwrap_or_default();
```

Comment:
I believe `list_cf()` is just opening in read-only under-the-hood. But, if someone just spun up a brand new node without a pre-existing rocksdb, subsequent code will create the rocksdb instance.

Reading on a yet-to-be-created rocksdb instance will fail, hence the `unwrap_or_default()`

Reply:
So I tested this out; I get the following when I pass `--ledger` as some random path:
```
[2023-11-27T21:04:50.440072617Z WARN  solana_ledger::blockstore_db] Unable to detect Rocks colums:
Error { message: "IO error: No such file or directory: While opening a file for sequentially reading:
/home/sol/disk2/rocksdb/CURRENT: No such file or directory" }
```
Upon further inspection, the bit after `to_cpath(path)?` can also result in an error:
```rust
            let ptr = ffi_try!(ffi::rocksdb_list_column_families(
                opts.inner,
                cpath.as_ptr(),
                &mut length,
            ));
```
That macro corresponds to:
```rust
macro_rules! ffi_try {
    ( $($function:ident)::*() ) => {
        ffi_try_impl!($($function)::*())
    };

    ( $($function:ident)::*( $arg1:expr $(, $arg:expr)* $(,)? ) ) => {
        ffi_try_impl!($($function)::*($arg1 $(, $arg)* ,))
    };
}

macro_rules! ffi_try_impl {
    ( $($function:ident)::*( $($arg:expr,)*) ) => {{
        let mut err: *mut ::libc::c_char = ::std::ptr::null_mut();
        let result = $($function)::*($($arg,)* &mut err);
        if !err.is_null() {
            return Err(Error::new($crate::ffi_util::error_message(err)));
        }
        result
    }};
}
```
https://github.com/rust-rocksdb/rust-rocksdb/blob/1c6ea8f81a2921594f05b2aae4f7cbd40ac89655/src/ffi_util.rs#L62-L81

So, I think we need the `unwrap_or_default()` to cover the case of a brand new blockstore being created (ie we don't want to fail on detecting a non-existent blockstore before we are able to create it)

Reply:
No worries. And sure thing, I'll just use a `match` statement to avoid any future ire in my/our direction 😉